### PR TITLE
Adjust normalization logic

### PR DESCRIPTION
**Pull request by alexkorotkikh**
Adjust query distance normalization logic to respect tiers configuration

### CODE REVIEW

File: src/main/java/org/kairosdb/core/http/rest/json/QueryParser.java [hunk 1]
```diff
@@ -451,6 +456,10 @@ else if (request.getStartRelative() != null)
 		{
 			throw new BeanValidationException(new SimpleConstraintViolation("start_time", "relative or absolute time must be set"), "query");
 		}
+
+		if (m_metricTiersConfig.getQueryDistanceHoursLimit() > 0) {
+			startTime = Math.max(startTime, now - m_metricTiersConfig.getQueryDistanceHoursLimit() * 60 * 60_000);
```
Comment (pitr) [inline src/main/java/org/kairosdb/core/http/rest/json/QueryParser.java:429]: hmm maybe `getQueryDistanceHoursLimit` could return `datapoints_ttl` by default, and then we wouldn't need the other `Math.max` call few lines later

File: src/main/java/org/kairosdb/core/http/rest/json/QueryParser.java [hunk 2]
```diff
@@ -104,10 +105,13 @@
...
 	private long datapoints_ttl = 3024000;
+	private MetricTiersConfiguration m_metricTiersConfig;
 
 	@Inject
-	public QueryParser(AggregatorFactory aggregatorFactory, GroupByFactory groupByFactory,
-	                   QueryPluginFactory pluginFactory)
+	public QueryParser(AggregatorFactory aggregatorFactory,
+					   GroupByFactory groupByFactory,
+	                   QueryPluginFactory pluginFactory,
+					   MetricTiersConfiguration metricTiersConfig)
```
Comment (pitr) [inline src/main/java/org/kairosdb/core/http/rest/json/QueryParser.java:82]: weird spacing

**Review by pitr — CHANGES_REQUESTED** [review]
left some comments

File: src/main/java/org/kairosdb/core/http/rest/json/QueryParser.java (hunk 1 above)
Comment (alexkorotkikh) [reply src/main/java/org/kairosdb/core/http/rest/json/QueryParser.java:429]: I don't think this is semantically correct. `datapoints_ttl` does not relate to distance limits

**Review by pitr — APPROVED** [review]

**Comment by alexkorotkikh** [comment]
👍

**Comment by pitr** [comment]
👍

**Pull request merged**